### PR TITLE
o/devicestate: fix panic when requesting a serial when device-service.access=offline

### PR DESCRIPTION
**Pull request by andrewphelpsj**
This fixes a panic that was reported when requesting a serial when `device-service.access=offline` is set.

### CODE REVIEW

File: overlord/devicestate/handlers_serial.go [hunk 1]
```diff
@@ -520,8 +522,7 @@ func getSerial(t *state.Task, regCtx registrationContext, privKey asserts.Privat
 	}
 
 	if !shouldRequest {
-		t.Logf("skipping getting serial, store is marked as offline")
-		return nil, nil, nil
```
Comment (pedronis) [inline overlord/devicestate/handlers_serial.go:524]: intersting that we didn't spot this, anyway in general one needs to be extra careful when returning nils without errors, it's a  contract that at most is viable for very close functions, which here is the case, but the caller was still unprepared and the situation untested

**Review by pedronis — APPROVED** [review]
thank you

**Review by ernestl — APPROVED** [review]
Thanks

**Review by MiguelPires — APPROVED** [review]
lgtm, thanks

**Pull request merged**